### PR TITLE
fix(core): honor property defaults instead of dropping them to null

### PR DESCRIPTION
**Pull request by halotukozak**
## Problem

A property that declares a JSON Schema `default` was being dropped. The parser marked every non-required property as nullable (`T? = null`), which discarded the declared default value entirely.

## Fix

A property with a `default` is not optional-null — it carries a concrete value, so it is now generated as a **non-nullable** field initialized to that default:

- scalar / enum / array defaults are honored
  - `mode: Mode = Mode.B`
  - `retries: Int = 3`
  - `tags: List<Tag> = listOf(Tag.X, Tag.Y)` / `emptyList()`
- array defaults arrive from the underlying parser as a Jackson `ArrayNode`; they are normalized to a plain `List` in the parser so the model layer stays Jackson-free
- object/map defaults are intentionally left unchanged for now (still nullable-null)

## Tests

New `SpecParserDefaultsTest` with a fixture covering scalar, enum, numeric, and array defaults, plus a property without a default (stays nullable). `:core:test` and `ktlintCheck` are green.

### CODE REVIEW

File: core/src/main/kotlin/com/avsystem/justworks/core/parser/SpecParser.kt [hunk 1]
```diff
@@ -502,6 +504,40 @@ object SpecParser {
...
+            else -> false
+        }
+
+    /**
+     * Normalizes a raw Swagger default into a plain Kotlin value the model layer can format
+     * without depending on Jackson. Array defaults arrive as a Jackson [ArrayNode]; unwrap them
+     * into a `List` of plain scalar values. Scalar defaults are already plain and pass through.
+     */
+    private fun normalizeDefault(default: Any?): Any? = when (default) {
+        is ArrayNode -> default.map { normalizeDefault(it) }
```
Comment (janwawszczak) [inline core/src/main/kotlin/com/avsystem/justworks/core/parser/SpecParser.kt:533]: Suggestion (worth fixing before merge): large int64 defaults silently lose precision in normalizeDefault. The chain                                                            
                                                                                                                                                                                    
```
   default.isInt -> default.intValue()                                                                                                                                            
   default.isNumber -> default.doubleValue()  
```                                                                                                                                      
                                                                                                                                                                                  
   skips LongNode (which has isInt() == false but isNumber() == true), so any default above 2^53 gets truncated when widened to Double. Realistic for OpenAPI specs declaring       
  format: int64 defaults (timestamps, Snowflake-style IDs, large counters). The failure mode is silent — no exception, no warning, just a wrong number in generated code.
                                                                                                                                                                                   
  One extra branch in the right order fixes it:                                                                                                                                  
 
```
   default.isBoolean -> default.booleanValue()                                                                                                                                      
   default.isInt -> default.intValue()
   default.isLong -> default.longValue()                                                                                                                                            
   default.isNumber -> default.doubleValue()  
```

File: core/src/main/kotlin/com/avsystem/justworks/core/gen/model/ModelGenerator.kt [hunk 2]
```diff
@@ -469,49 +470,67 @@ internal object ModelGenerator {
...
 
-                else -> throw IllegalArgumentException("Unsupported default value type: ${prop.type}")
+                else -> throw IllegalArgumentException("Unsupported default value type: $type")
             }
         }
 
         is TypeRef.Reference -> {
-            val constantName = prop.defaultValue.toString().toEnumConstantName()
-            CodeBlock.of("%T.%L", ClassName(hierarchy.modelPackage, prop.type.schemaName), constantName)
+            val constantName = value.toString().toEnumConstantName()
```
Comment (janwawszczak) [inline core/src/main/kotlin/com/avsystem/justworks/core/gen/model/ModelGenerator.kt:544]: Suggestion — coverage gap: the new Array branch (this listOf(...) / emptyList() logic) doesn't have a direct test. The four tests in SpecParserDefaultsTest stop at the parser   
  layer — they assert PropertyModel.nullable and .defaultValue, but never verify what formatDefaultValue actually emits. A regression here (e.g. someone swapping joinToCode for   
  joinToString and breaking type-name resolution, or mis-recursing on type.items) would slip through unnoticed.
                                                                                                                                                                                    
  The natural home is ModelGeneratorTest.kt (next to optional property is nullable with default null at line ~89), which already exercises the generator end-to-end. A pair of   
   tests for the boundary cases:                                                                                                                                                    
                                                                                                                                                                                  
```
  @Test                                                                                                                                                                            
   fun `array property with default emits listOf with element literals`() {
       // generate from a spec with: tags: { type: array, items: { type: string }, default: ["X", "Y"] }                                                                            
       // assert the generated property reads: val tags: List<String> = listOf("X", "Y")                                                                                          
   }                                         
                                                                                                                                                                                    
   @Test                                                                                                                                                                            
   fun `array property with empty default emits emptyList`() {                                                                                                                      
       // generate from a spec with: tags: { type: array, items: { type: string }, default: [] }                                                                                    
       // assert the generated property reads: val tags: List<String> = emptyList()                                                                                                 
   }                                                                                                                                                                              
          
```                                                                                                                                                                          
   Both branches of the if (elements.isEmpty()) get covered, plus the joinToCode happy path. Non-blocker, but closes the loop end-to-end and would have caught a Jackson-leakage bug
    at integration time.

**Review by janwawszczak — APPROVED** [review]
Approve, two suggestions to consider :)

File: core/src/main/kotlin/com/avsystem/justworks/core/parser/SpecParser.kt (hunk 1 above)
Comment (halotukozak) [reply core/src/main/kotlin/com/avsystem/justworks/core/parser/SpecParser.kt:533]: I've added even more types
